powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/ActivationHistoryEntity.java

```diff
@@ -216,54 +193,31 @@ public void setActivationVersion(Integer version) {
         this.activationVersion = version;
     }
 
+    public String getActivationName() {
+        return activationName;
+    }
+
+    public void setActivationName(final String activationName) {
+        this.activationName = activationName;
+    }
+
     @Override
     public int hashCode() {
-        int hash = 7;
-        hash = 23 * hash + Objects.hashCode(this.activation);
-        hash = 23 * hash + Objects.hashCode(this.activationStatus);
-        hash = 23 * hash + Objects.hashCode(this.eventReason);
-        hash = 23 * hash + Objects.hashCode(this.externalUserId);
-        hash = 23 * hash + Objects.hashCode(this.timestampCreated);
-        hash = 23 * hash + Objects.hashCode(this.activationVersion);
-        return hash;
+        return Objects.hash(id);
     }
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null) {
+        if (null == obj) {
             return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
-        final ActivationHistoryEntity other = (ActivationHistoryEntity) obj;
-        if (!Objects.equals(this.activation, other.activation)) {
-            return false;
-        }
-        if (!Objects.equals(this.activationStatus, other.activationStatus)) {
-            return false;
-        }
-        if (!Objects.equals(this.eventReason, other.eventReason)) {
-            return false;
-        }
-        if (!Objects.equals(this.externalUserId, other.externalUserId)) {
-            return false;
-        }
-        if (!Objects.equals(this.timestampCreated, other.timestampCreated)) {
+        } else if (this == obj) {
+            return true;
+        } else if (!this.getClass().equals(ProxyUtils.getUserClass(obj))) {
             return false;
+        } else {
+            final ActivationHistoryEntity that = (ActivationHistoryEntity) obj;
+            return null != this.getId() && this.getId().equals(that.getId());
```

Comment:
This change is incorrect, because an unpersisted and persisted entity from the same instance will not be equal. This is caused by autegenerated ID, which is initially null and then becomes non-null. This can bring unexpected and hard to discover issues when using collections.